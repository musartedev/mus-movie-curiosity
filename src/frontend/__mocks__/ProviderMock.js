import React from 'react';
import { Router } from 'react-router';
import { createBrowserHistory } from 'history';
import { BrowserRouter } from 'react-router-dom';
import { SearchContextProvider } from '../context/Search';
import { SortContextProvider } from '../context/Sort';

const history = createBrowserHistory();

const ProviderMock = ({ children }) => (
  <Router history={history}>
    <SearchContextProvider>
      <SortContextProvider>
        <BrowserRouter>{children}</BrowserRouter>
      </SortContextProvider>
    </SearchContextProvider>
  </Router>
);

export default ProviderMock;
