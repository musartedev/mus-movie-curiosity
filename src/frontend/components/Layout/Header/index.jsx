import React from 'react';
import NavList from '../../common/NavList';
import { headerNav } from '../../../constants/Nav';

import Logo from '../../../assets/img/Logo.png';

import '../../../styles/components/Layout/Header.styl';

const Header = () => {
  return (
    <header className="Header">
      <div className="Header__brand">
        <img src={Logo} alt="Movie Curiosity Logo" />
      </div>
      <div className="Header__nav">
        <NavList navItems={headerNav} horizontal />
      </div>
    </header>
  );
};

export default Header;
