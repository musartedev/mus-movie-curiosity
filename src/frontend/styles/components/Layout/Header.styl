
@import '../variables.styl'

.Header
  display grid
  grid-template-columns auto 1fr
  align-content center
  padding 10px $layout-padding
  background-color $background
  transition all 1s
  font-size 1em

  &__brand
    align-self center
    img
      height 80px

  &__nav
    justify-self end

  @media (min-width $device-md)
    font-size 1.5em

