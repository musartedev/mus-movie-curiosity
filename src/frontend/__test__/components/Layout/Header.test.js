import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import ProviderMock from '../../../__mocks__/ProviderMock';
import Header from '../../../components/Layout/Header';

describe('<Header />', () => {
  test('Header component should render', () => {
    const header = mount(
      <ProviderMock>
        <Header />
      </ProviderMock>
    );

    expect(header.length).toEqual(1);
  });

  test('Header snapshot should match', () => {
    const header = create(
      <ProviderMock>
        <Header />
      </ProviderMock>
    );

    expect(header.toJSON()).toMatchSnapshot();
  });
});
