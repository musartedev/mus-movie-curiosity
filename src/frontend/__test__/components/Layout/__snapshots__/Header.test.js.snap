// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Header snapshot should match 1`] = `
<header
  className="Header"
>
  <div
    className="Header__brand"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Movie Curiosity Logo"
        src=""
      />
    </a>
  </div>
  <div
    className="Header__nav"
  >
    <div
      className="nav-list nav-list--horizontal"
    >
      <nav>
        <ul>
          <li
            className="nav-list__item"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li
            className="nav-list__item"
          >
            <a
              href="/trending"
              onClick={[Function]}
            >
              Trending
            </a>
          </li>
          <li
            className="nav-list__item"
          >
            <a
              href="/now-playing"
              onClick={[Function]}
            >
              Latest
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
